js/main.js

```diff
@@ -379,12 +380,16 @@ $(document).ready(function() {
     legend.onAdd = function () { return L.DomUtil.get('legend'); };
 
     // Populate dropdown
-    loadCityIDs().done(function(cityIDs) {
-        for (var i = 0; i < cityIDs.length; i++) {
-            var id = cityIDs[i];
+    loadCities().done(function(cities) {
+
+        // cache the results
+        cityDirectory = {};
+        for (var i = 0; i < cities.length; i++) {
```

Comment:
Initialize length once instead of in each loop: `for (var i = 0, n = cities.length; i < n; i++)`